### PR TITLE
Updated HMNFD design docs.

### PR DESCRIPTION
**Pull request by mpkelly-hpe**
Updated HMNFD design docs.

### CODE REVIEW

File: HMNFD_Design.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+The Shasta State Change Notification (SCN) fanout service provides the ability
+to notify subscribers of component hardware state changes and other changes
+made to and by the Hardware State Manager.  _hmnfd_ is a companion service
+to the Hardware State Manager in that it does the heavy lifting of 
+distributing state changes and managing subscriptions.
+
+This service uses a RESTful interface to provide the following functions:
+
+* Subscription for component state changes:
+  * Hardware state (On, Off, Ready, etc.)
```
Comment (nieuwsma) [inline HMNFD_Design.md:12]: is Ready, really a hardware state?

File: HMNFD_Design.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  * Logical state (arbitrary, e.g., AdminDown)
+  * Role (Compute, NCN, etc.)
+  * Enabled state
+  * Flag (OK, Alert, Warning, etc.)
+* Ability to view current subscriptions
+* Ability to delete subscriptions
+* Ability to get and modify current service operating parameters
+* Ability to receive State Change Notifications from the Hardware State Manager
+
+_hmnfd_ typically runs on an SMS cluster as one or more Docker container
```
Comment (nieuwsma) [inline HMNFD_Design.md:22]: I dont think SMS is a thing anymore.   I think I would say: kubernetes cluster in the services mesh

File: HMNFD_Design.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+### Health, Liveness, and Readiness
+
+HMNFD has health, liveness, and readiness APIs which are used by Kubernetes
+to determine the fitness of HMNFD.  Various things are checked, such as
+ETCD connectivity, etc. to determine fitness and health.
+
+## hmnfd API
+
+_hmnfd_'s RESTful API is a follows:
```
Comment (nieuwsma) [inline HMNFD_Design.md:117]: if all of this detail is in the swagger I think we should point at the swagger and remove it from here, that way there is only 1 place to keep up.

**Review by nieuwsma — CHANGES_REQUESTED** [review]

File: HMNFD_Design.md (hunk 1 above)
Comment (mpkelly-hpe) [reply HMNFD_Design.md:12]: Technically not but it is lumped into the HW state by HSM.  The Logical State field is not where Ready
is ever held.

File: HMNFD_Design.md (hunk 2 above)
Comment (mpkelly-hpe) [reply HMNFD_Design.md:22]: Fixed.

File: HMNFD_Design.md (hunk 3 above)
Comment (mpkelly-hpe) [reply HMNFD_Design.md]: Done.

**Review by shunr-hpe — APPROVED** [review]

**Review by nieuwsma — APPROVED** [review]

**Pull request merged**